pages/snake.js

```diff
@@ -1,57 +1,70 @@
-import { useEffect, useRef } from 'react';
+import { useEffect, useState, useRef } from 'react';
 import { HeadLine } from "../components/Header";
-import { Matchfield } from "../components/Snake";
-import { Main } from '../components/XOGame';
+import { Matchfield, Score } from "../components/Snake";
+import { Container, Menu } from '../components/Memory';
+import Link from 'next/link';
 
 export default function Snake() {
     const canvasRef = useRef(null);
 
     let directionX = 10;
     let directionY = 0;
 
+    let foodX;
+    let foodY;
+
     const board_border = 'black';
     const board_background = 'white';
     const snake_col = 'lightblue';
     const snake_border = 'darkblue';
 
-    function drawSnake(snake, snakeboard_ctx) {
-        snake.forEach((snakePart) => drawSnakePart(snakePart, snakeboard_ctx));
+    const [score, setScore] = useState(0);
+
+    function drawSnake(snake, snakeboardCTX) {
+        snake.forEach((snakePart) => drawSnakePart(snakePart, snakeboardCTX));
     }
 
-    function drawSnakePart(snakePart, snakeboard_ctx) {
-        snakeboard_ctx.fillStyle = snake_col;
-        snakeboard_ctx.strokeStyle = snake_border;
-        snakeboard_ctx.fillRect(snakePart.x, snakePart.y, 10, 10);
-        snakeboard_ctx.strokeRect(snakePart.x, snakePart.y, 10, 10);
+    function drawSnakePart(snakePart, snakeboardCTX) {
+        snakeboardCTX.fillStyle = snake_col;
+        snakeboardCTX.strokeStyle = snake_border;
+        snakeboardCTX.fillRect(snakePart.x, snakePart.y, 10, 10);
+        snakeboardCTX.strokeRect(snakePart.x, snakePart.y, 10, 10);
     }
 
-    function clearCanvas(snakeboard, snakeboard_ctx) {
-        snakeboard_ctx.fillStyle = board_background;
-        snakeboard_ctx.strokestyle = board_border;
-        snakeboard_ctx.fillRect(0, 0, snakeboard.width, snakeboard.height);
-        snakeboard_ctx.strokeRect(0, 0, snakeboard.width, snakeboard.heigth);
+    function clearCanvas(snakeboard, snakeboardCTX) {
+        snakeboardCTX.fillStyle = board_background;
+        snakeboardCTX.strokestyle = board_border;
+        snakeboardCTX.fillRect(0, 0, snakeboard.width, snakeboard.height);
+        snakeboardCTX.strokeRect(0, 0, snakeboard.width, snakeboard.heigth);
     }
 
-    function moveSnake(snake, directionX, directionY) {
+    function moveSnake(snake, directionX, directionY, snakeboard, score) {
         const head = { x: snake[0].x + directionX, y: snake[0].y + directionY };
         snake.unshift(head);
-        snake.pop();
+        const has_eaten_food = snake[0].x === foodX && snake[0].y === foodY;
```

Comment:
I think it's not common to name constants as questions. Maybe name it `eatFood` or `foodPosition` `hastEaten` or `eatenFood`